pkg/operator/controller/garden/access/add.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package access
+
+import (
+	"github.com/spf13/afero"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+)
+
+// ControllerName is the name of this controller.
+const ControllerName = "garden-access"
+
+// AddToManager adds Reconciler to the given manager.
+func (r *Reconciler) AddToManager(mgr manager.Manager, namespace, secretName string) error {
+	if r.Client == nil {
+		r.Client = mgr.GetClient()
+	}
+
+	if r.Manager == nil {
+		r.Manager = mgr
+	}
+
+	if r.FS == nil {
+		r.FS = afero.NewOsFs()
+	}
+
+	accessSecretPredicates := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		_, hasRenewAnnotation := o.GetAnnotations()[resourcesv1alpha1.ServiceAccountTokenRenewTimestamp]
```

Comment:
Wouldn't it make more sense to check here if the `token` key is populated in the `data`? You check this anyways in the reconciler, but I guess it's more appropriate here in the predicate already.

Reply:
Hm, is it really? I used this check here because it's very cheap compared to extracting and unmarshalling the kubeconfig. 

I even think it's fine to ignore this in the predicate (hence, remove it) and only do it in the reconciler, where we need the token anyway. It's only relevant at the very beginning of the Garden creation.

Reply:
I missed that this access secret does not only contain a token, but a full kubeconfig, and the token is populated directly into it. In this light, of course, you're right - let's keep it like this for pragmatic reasons.